src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDataReader.cs

```diff
@@ -1020,7 +1025,7 @@ private Type GetProviderSpecificFieldTypeInternal(_SqlMetaData metaData)
 
                 if (metaData.type == SqlDbType.Udt)
                 {
-                    throw UdtNotSupportedException();
+                    providerSpecificFieldType = MetaType.MetaMaxVarBinary.SqlType;
```

Comment:
I'll revert this. In my original implementation, I converted everything to use binary and missed reverting this. I'll update to throw here instead of in GetProviderSpecificFieldType.